ytmusicapi/parsers/search.py

```diff
@@ -34,8 +34,17 @@ def parse_top_result(data, search_result_types):
             search_result["videoType"] = nav(on_tap, NAVIGATION_VIDEO_TYPE)
 
     if result_type in ["song", "video", "album"]:
+        on_tap = data.get("onTap")
+        if on_tap is not None:
+            if "watchEndpoint" in on_tap:
+                if nav(on_tap, WATCH_VIDEO_ID) is not None:
+                    search_result["videoId"] = nav(on_tap, WATCH_VIDEO_ID)
+            if "navigationEndpoint" in on_tap:
+                if nav(on_tap, NAVIGATION_VIDEO_TYPE) is not None:
+                    search_result["videoType"] = nav(on_tap, NAVIGATION_VIDEO_TYPE)
```

Comment:
Instead of triple-if, just navigate directly to the key you want to access using `nav`. Make sure to set `none_if_absent` (i.e. `nav(data, keys, True)`) - this way it'll just error out with `None` at the first missing key in the chain of `keys`

```suggestion
       search_result["videoId"] = nav(data, ["onTap"] + WATCH_VIDEO_ID, True)
       search_result["videoType"] = nav(data, ["onTap"] + NAVIGATION_VIDEO_TYPE, True)
```

Reply:
Sure I can do that

Reply:
I made this change, thank you!